host/cli/fix.go

```diff
@@ -0,0 +1,607 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-docopt"
+	"github.com/flynn/flynn/Godeps/_workspace/src/gopkg.in/inconshreveable/log15.v2"
+	"github.com/flynn/flynn/appliance/postgresql/client"
+	pgstate "github.com/flynn/flynn/appliance/postgresql/state"
+	"github.com/flynn/flynn/controller/client"
+	ct "github.com/flynn/flynn/controller/types"
+	"github.com/flynn/flynn/controller/utils"
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/host/types"
+	"github.com/flynn/flynn/pkg/cluster"
+)
+
+func init() {
+	Register("fix", (&clusterFixer{}).Run, `
+usage: flynn-host fix [options]
+
+Attempts to fix a broken cluster by starting missing jobs.
+
+Options:
+    -n, --min-hosts=<n>  minimum expected number of hosts (required)
+	--peer-ips=<iplist>  list of host IPs (required if discoverd is down)
+`)
+}
+
+type clusterFixer struct {
+	hosts []*cluster.Host
+	c     *cluster.Client
+	l     log15.Logger
+}
+
+func (f *clusterFixer) Run(args *docopt.Args, c *cluster.Client) error {
+	f.c = c
+	f.l = log15.New()
+	var err error
+
+	minHosts, err := strconv.Atoi(args.String["--min-hosts"])
+	if err != nil || minHosts < 1 {
+		return fmt.Errorf("invalid or missing --min-hosts value")
+	}
+
+	f.hosts, err = c.Hosts()
+	if err != nil {
+		f.l.Error("unable to list hosts from discoverd, falling back to peer IP list", "error", err)
+		var ips []string
+		if ipList := args.String["--peer-ips"]; ipList != "" {
+			ips = strings.Split(ipList, ",")
+			if minHosts == 0 {
+				minHosts = len(ips)
+			}
+		}
+		if len(ips) == 0 {
+			return fmt.Errorf("error connecting to discoverd, use --peer-ips: %s", err)
```

Comment:
Oh unless you want it there, didn't realise this was in an `if err != nil` block